libexec/rtld-elf/rtld_paths.h

```diff
@@ -74,6 +77,18 @@
 #define	LD_			"LD_64_"
 #endif
 
+#ifndef _PATH_ELF_C18N_HINTS
+#define	_PATH_ELF_C18N_HINTS	"/var/run/ld-elf.so.hints"
```

Comment:
Does this find the right libthr?

Reply:
I admit on seeing this I had wondered casually if the string was intended to be ```/var/run/ld-elf-c18n.so.hints``` so that a different hints file was used as compared to the baseline ABI.

Reply:
Ugh, ldconfig_paths doesn't include /lib and /usr/lib but rc.d/ldconfig implicitly adds them to the front (in contrast with ldconfig32_paths which puts them in the config file rather than the rc file, and ldconfig64_paths which puts them neither and thus relies on rtld's fallback). Since hints are searched before the default search path this means they likely are picked up from the wrong place now, I think :(

Also means things are slightly dodgy for ld-elf64, as /usr/lib64 comes after /usr/local64/lib in the search order... so I guess that explains why ldconfig32_paths has /usr/lib32 in it. Why couldn't it be consistent with ldconfig_paths...

Reply:
@brooksdavis I would guess that it does as otherwise cheribsdtest-mt-c18n would have failed with certain TLS related tests that use pthread_create, but more investigation is needed to confirm this.

Reply:
@rwatson This line was initially `/var/run/ld-elf-c18n.so.hints` as committed by @brooksdavis but since that file didn’t exist, I changed it to the current line to obtain a quick fix that seems to work.

Reply:
@brooksdavis Actually, this doesn't find the right `libthr`, and I made a mistake while testing which resulted in the issue not showing up.

What would be the best way to add `ld-elf-c18n.so.hints`? It seems to me that the solution is to add some code to `rc.d/ldconfig`, but I'm not sure what the best practice is.

Reply:
Adding another block like the one for ldconfig32 in rc.d/ldconfig is probably the right thing.